test/boost/schema_change_test.cc

```diff
@@ -664,45 +664,45 @@ future<> test_schema_digest_does_not_change_with_disabled_features(sstring data_
 
 SEASTAR_TEST_CASE(test_schema_digest_does_not_change) {
     std::vector<utils::UUID> expected_digests{
-        utils::UUID("492719e5-0169-30b1-a15e-3447674c0c0c"),
-        utils::UUID("be3c0af4-417f-31d5-8e0e-4ac257ec00ad"),
-        utils::UUID("be3c0af4-417f-31d5-8e0e-4ac257ec00ad"),
-        utils::UUID("be3c0af4-417f-31d5-8e0e-4ac257ec00ad"),
-        utils::UUID("be3c0af4-417f-31d5-8e0e-4ac257ec00ad"),
-        utils::UUID("4198e26c-f214-3888-9c49-c396eb01b8d7"),
-        utils::UUID("5c9cadec-e5df-357e-81d0-0261530af64b"),
-        utils::UUID("1d91ad22-ea7c-3e7f-9557-87f0f3bb94d7"),
-        utils::UUID("2dcd4a37-cbb5-399b-b3c9-8eb1398b096b")
+        utils::UUID("8182496e-4baf-3a07-91e6-caa140388846"),
+        utils::UUID("a65ea746-4d8a-3e5c-8fbf-5f70c14dbcbc"),
+        utils::UUID("a65ea746-4d8a-3e5c-8fbf-5f70c14dbcbc"),
+        utils::UUID("4c138336-4677-3520-8556-4aab007cfedb"),
+        utils::UUID("4c138336-4677-3520-8556-4aab007cfedb"),
+        utils::UUID("62e1e586-6eec-3ff5-882a-89386664694b"),
+        utils::UUID("daf6ded5-c294-3b07-b6a0-1b318a3c2e17"),
+        utils::UUID("370c7d8e-0a4a-394d-b627-318805c64584"),
+        utils::UUID("74bcc842-6a3f-35cd-84ea-3b330ed83c4c")
     };
```

Comment:
This test is intended to ensure the schema does NOT change. Not that it changed. You're hard-coding an upgrade failure here.

Reply:
I explained in the commit why it's fine to add a new table to system_distributed without protecting it by a schema feature:
https://github.com/scylladb/scylla/pull/5485/commits/52f6d4787e20cd9c8f561e0fc1bc404226e0112a

That's why I had to upgrade the digests and sstables -- because the test detected a new table.

Is my explanation wrong?

Reply:
And there is no upgrade failure.

Reply:
No, the explanation is correct (I think), but let's think of a way to ensure that the test isn't hiding a schema breakage.

Reply:
The test is there to make sure that if we upgrade a subset of nodes in a cluster, the subset reaches schema agreement with the rest of the nodes. A comment explains it:
```
// We don't want schema digest to change between Scylla versions because that results in a schema disagreement
// during rolling upgrade.
```

However, the test is a bit too specific, i.e. it sometimes introduces false positives (by "positive" I mean that the test "catches a bug").

This happens when:
1. the upgraded subset does have a different schema, **but**
2. the subset also brings the rest of the nodes up-to-date, so nothing bad actually happened.

The test catches 1., but it doesn't know about 2.

My particular situation here is that the upgraded subset announces a new distributed table. This indeed introduces a schema change (1.), but that's an `announce_new_column_family`, so other nodes follow up (2.), and we get schema agreement.

This happens *by design of the test*. The test author (@tgrabiec) predicted that it would!
```
    const bool regenerate = false;
```

and fortunately, @psarna added a comment which explains what the flag is for:
```
    // NOTICE: Regenerating data for this test may be necessary when a system table is added.
    // This test uses pre-generated sstables and relies on the fact that they are up to date
    // with the current system schema. If it is not, the schema will be updated, which will cause
    // new timestamps to appear and schema digests will not match anymore.
```

This `regenerate` flag exists here for us to use it when necessary. I used it correctly.

There only way to
> ensure that the test isn't hiding a schema breakage

would be to design a new test, which does something completely different. That is out of scope of this issue.

Reply:
Right. A way to verify this is to run the test (unmodified) on the previous commit, just to be sure previous commits didn't introduce breakages that _this_ commit enshrines.

Reply:
I checked that, it passes.